File: Medium/75_sort_colors/solution.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nums[:] = [0]*nums.count(0) + [1]*nums.count(1) + [2]*nums.count(2)

        # zeros, ones, twos = nums.count(0), nums.count(1), nums.count(2)
        # for i in range(len(nums)):
        #     if zeros: 
        #         nums[i] = 0
        #         zeros -= 1
        #     elif ones: 
        #         nums[i] = 1
        #         ones -= 1
        #     else: 
        #         nums[i] = 2
        #         twos -= 1

File: Medium/75_sort_colors/solution.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int zeros = 0;
        for (int n : nums) {
            if (n == 0) {
                zeros++;
            }
        }
        int ones = 0;
        for (int n : nums) {
            if (n == 1) {
                ones++;
            }
        }
        int twos = 0;
        for (int n : nums) {
            if (n == 2) {
                twos++;
            }
        }

        for (int i=0; i<nums.size(); i++) {
            if (zeros) {
                nums[i] = 0;
                zeros--;
            }
            else if (ones) {
                nums[i] = 1;
                ones--;
            }
            else {
                nums[i] = 2;
                twos--;
            }
        }
    }
};
